recsplit/eliasfano32/elias_fano.go

```diff
@@ -305,8 +305,12 @@ func ReadEliasFano(r []byte) (*EliasFano, int) {
 	ef.count = binary.BigEndian.Uint64(r[:8])
 	ef.u = binary.BigEndian.Uint64(r[8:16])
 	ef.maxOffset = ef.u - 1
-	p := (*[maxDataSize / 8]uint64)(unsafe.Pointer(&r[16]))
-	ef.data = p[:]
+
+	ef.data = make([]uint64, len(r[16:])/8)
```

Comment:
This needs to be undone, because the idea is to use memory-mapped files and not allocate it on the heap

Reply:
we discussed that already, using unsafe pointer `p` leads to use-after-close in indexes